### ISSUE TITLE
Add entry_points to setuptools’ configuration to provide command-line console scripts

### ISSUE THREAD

**Issue by vitorsr** [pymupdf/PyMuPDF#3199]
Currently, PyMuPDF is not pipx-installable because it does not provide any console commands:

```
$ pipx install PyMuPDF

No apps associated with package pymupdf or its dependencies. If you are attempting to install a library, pipx should
not be used. Consider using pip or a similar tool instead.
```

To make, e.g., fitz, directly available from the command-line we need to add entry_points to setuptools’ configuration [1]:

```py
        entry_points = {"console_scripts": ["fitz = fitz.__main__:main"]},
```

This would require modifying the pipcl.Package class.

It could alternatively be added to pyproject.toml [2]:

```diff
diff --git a/pyproject.toml b/pyproject.toml
index 10c9a43..05b7117 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -5,3 +5,6 @@ requires = ["libclang", "swig", "setuptools", "psutil"]
 #
 build-backend = "setup"
 backend-path = ["."]
+
+[project.scripts]
+fitz = fitz.__main__:main
```

[1] https://setuptools.pypa.io/en/latest/userguide/entry_point.html

[2] https://packaging.python.org/en/latest/specifications/pyproject-toml/#entry-points

**Comment by julian-smith-artifex-com** [comment]
Thanks for this, i didn't know about entry_points before.

In my tree i have added support for entry_points to pipcl.py and modified setup.py, so that installation gives a `pymupdf` command that runs fitz/__main__.py.

Depending on internal review, it might get included in the next release.

Thanks again.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.26.

**Comment by vitorsr** [comment]
Thanks, Julian! Really appreciate all the good work on (Py-) MuPDF! Regards.
